src/fileio.py

```diff
@@ -111,52 +112,52 @@ def write_file_header(self, midifile, pattern):
                             pattern.format, 
                             len(pattern),
                             pattern.resolution)
-        midifile.write('MThd%s' % packdata)
+        midifile.write(b'MThd' + packdata)
             
     def write_track(self, midifile, track):
-        buf = ''
+        buf = b''
         self.RunningStatus = None
         for event in track:
             buf += self.encode_midi_event(event)
         buf = self.encode_track_header(len(buf)) + buf
```

Comment:
This applies to the original code too: calling += in a loop might lead to O(N**2) running time.  It would be more efficient to collect event representations into a list and combine them at the end with `b''.join()`.